lib/huginn_acumen_order_agent/acumen_order_agent.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+module Agents
+    class AcumenOrderAgent < Agent
+        include WebRequestConcern
+        include AcumenOrderQueryConcern
+        include InvoiceQueryConcern
+        include InvoiceDetailQueryConcern
+
+        default_schedule '12h'
+
+        can_dry_run!
+        default_schedule 'never'
+
+        description <<-MD
+        Huginn agent for retrieving sane ACUMEN invoice data.
+
+        ## Agent Options
+        The following outlines the available options in this agent
+
+        ### Acumen Connection
+        * endpoint: The root URL for the Acumen API
+        * site_code: The site code from Acumen
+        * password: The Acumen API password
+
+        ### Payload Status
+
+        `status: 200`: Indicates a true success. The agent has output the full
+        range of expected data.
+
+        `status: 206`: Indicates a partial success. The products within the bundle
+        are vaild, but the bundle _may_ be missing products that were somehow invalid.
+
+        `status: 500`: Indicates a processing error. This may represent a complete
+        process failure, but may also be issued in parallel to a `202` payload.
+
+        Because this agent receives an array of Order Codes as input, errors will be issued in
+        such a way that product processing can recover when possible. Errors that occur within
+        a specific product bundle will emit an error event, but the agent will then move
+        forward processing the next bundle.
+
+        For example, if this agent receives two products as input (`A` and `B`), and we fail to
+        load the Inv_Product record for product `A`, the agent would emit an error payload of:
+
+        ```
+        {
+          status: 500,
+          scope: 'Fetch Inv_Product Data',
+          message: 'Failed to lookup Inv_Product record for Product A',
+          data: { product_id: 123 },
+          trace: [ ... ]
+        }
+        ```
+
+        The goal of this approach is to ensure the agent outputs as much data as reasonably possible
+        with each execution. If there is an error in the Paperback version of a title, that shouldn't
+        prevent this agent from returning the Hardcover version.
+
+        MD
+
+        def default_options
+            {
+                'endpoint' => 'https://example.com',
+                'site_code' => '',
+                'password' => '',
+                'digital_formats' => [],
+                'attribute_to_property' => {},
```

Comment:
You have digital_formats and attribute_to_property default here as default options but they don't seem to be used anywhere

Reply:
Yeah. Those need to go. Fixed